plugins/tpm/fu-tpm-v2-device.c

```diff
@@ -222,6 +222,55 @@ fu_tpm_v2_device_setup_pcrs(FuTpmV2Device *self, ESYS_CONTEXT *ctx, GError **err
 	return TRUE;
 }
 
+static gboolean
+fu_tpm_v2_device_ensure_commands(FuTpmV2Device *self, ESYS_CONTEXT *ctx, GError **error)
+{
+	gboolean seen_upgrade_data = FALSE;
+	gboolean seen_upgrade_start = FALSE;
+	TSS2_RC rc;
+	g_autofree TPMS_CAPABILITY_DATA *capability = NULL;
+	g_autoptr(GString) str = g_string_new(NULL);
+
+	rc = Esys_GetCapability(ctx,
+				ESYS_TR_NONE,
+				ESYS_TR_NONE,
+				ESYS_TR_NONE,
+				TPM2_CAP_COMMANDS,
+				TPM2_CC_FIRST,
+				TPM2_MAX_CAP_CC,
+				NULL,
+				&capability);
+	if (rc != TSS2_RC_SUCCESS) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_NOT_SUPPORTED,
+				    "capability request failed for TPM2_CAP_COMMANDS");
+		return FALSE;
+	}
+	for (guint i = 0; i < capability->data.ppCommands.count; i++) {
+		guint cap_cmd = capability->data.ppCommands.commandCodes[i] & 0xFFFF;
+
+		if (str->len > 0)
+			g_string_append(str, ", ");
+		g_string_append_printf(str, "0x%04x", cap_cmd);
+
+		/* ones we care about */
+		if (cap_cmd == TPM2_CC_FieldUpgradeStart) {
+			seen_upgrade_start = TRUE;
+		} else if (cap_cmd == TPM2_CC_FieldUpgradeData) {
+			seen_upgrade_data = TRUE;
+		}
+	}
+	g_debug("CAP_COMMANDS: %s", str->str);
+
+	/* both available */
+	if (seen_upgrade_start && seen_upgrade_data)
+		fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
```

Comment:
presumably we actually need a routine that can do updates to add this flag

Reply:
Yup, I've just added this although it's 100.00% untested.